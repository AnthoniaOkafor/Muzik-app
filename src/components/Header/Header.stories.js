import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withKnobs, text, number } from "@storybook/addon-knobs";
import Header from "./index";

storiesOf("Header", module)
  .addDecorator(
    withInfo({
      text: "This is a header component"
    })
  )
  .addDecorator(withKnobs())
  .add("Header", () => (
    <Header
      heading={text("heading", "Sample text")}
      size={number("size", "2")}
      color={text("color", "green")}
    />
  ));
