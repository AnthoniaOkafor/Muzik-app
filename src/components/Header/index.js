import React from "react";

const Header = ({heading, color, size}) => {
    if (size === 3) {
        return(
            <div className="header">
                <h3 color={color}>{heading}</h3>
            </div>
        )
    }else if (size === 2) {
        return(
            <div className="header">
                <h2 color={color}>{heading}</h2>
            </div>
        )
    }else 
        return(
            <div className="header">
                <h1 color={color}>{heading}</h1>
            </div>
        )
}

export default Header;