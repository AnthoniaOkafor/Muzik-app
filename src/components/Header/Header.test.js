import React from "react";
import { shallow } from 'enzyme';

import Header from "./index";

describe('Header snapshot for different level heading', () => {
    it('should match the H1 snapshot', () => {
        const wrapper = shallow(
            <Header 
                heading= 'h1 heading'
                color = "blue"
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should match the H2 snapshot', () => {
        const wrapper = shallow(
            <Header 
                heading= 'h2 heading'
                color = "blue"
                size = {2}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    it('should match the H3 snapshot', () => {
        const wrapper = shallow(
            <Header 
                heading= 'h3 heading'
                color = "blue"
                size = {3}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
    
})

describe('Header output', () =>{

    it('renders the title in <h1> tags', () => {
        const header = shallow(<Header heading='show some heading' />);
        const title = <h1>show some heading</h1>;
    
        expect(header.contains(title)).toEqual(true);
    });
    it('renders the title in <h2> tags', () => {
        const header = shallow(<Header heading='show some heading' size={2}/>);
        const title = <h2>show some heading</h2>;
    
        expect(header.contains(title)).toEqual(true);
    });
    it('renders the title in <h3> tags', () => {
        const header = shallow(<Header heading='show some heading' size={3}/>);
        const title = <h3>show some heading</h3>;
    
        expect(header.contains(title)).toEqual(true);
    });
    // it('renders a heading element', () =>{
    //     const header = shallow(<Header heading='try this to see'/>)
    //     expect(header.text()).toBe('try this to see')
    // })    
})

