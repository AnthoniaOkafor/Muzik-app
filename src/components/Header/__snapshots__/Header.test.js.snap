// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot for different level heading should match the H1 snapshot 1`] = `ShallowWrapper {}`;

exports[`Header snapshot for different level heading should match the H2 snapshot 1`] = `ShallowWrapper {}`;

exports[`Header snapshot for different level heading should match the H3 snapshot 1`] = `ShallowWrapper {}`;
